### PR TITLE
Completed task 2. Needs reviewing

### PR DESCRIPTION
**Pull request by denton909**
I have completed task 2. Please could someone review this for me.

Thanks

Phil

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const seed = require('../db/seeds/seed');
+const testData = require('../db/data/test-data/index');
+const db = require('../db/connection');
+const request = require("supertest");
+const app = require("../app")
+
+beforeEach(() => seed(testData));
+afterAll(() => db.end())
+
```
Comment (Dutton1990) [inline __tests__/app.test.js:9]: Nice implicit returning here! ☝️

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+describe("app test suite", () => {
+    test("200 GET all topics at endpoint of /api/topics and return array of all topic objects", () => {
+        return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.topics).toHaveLength(3);
```
Comment (Dutton1990) [inline __tests__/app.test.js:16]: Very nice to have the length check

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        })
+    })
+    test("404 responds with an error message when passed the wrong endpoint", () => {
+        return request(app)
+        .get("/api/banana")
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Data Not Found");
+        })
+    })
```
Comment (Dutton1990) [inline __tests__/app.test.js:30]: This error test isn't specific to this endpoint, in theory going to /api/banana is the same as going to /nonsense - it's a path that doesn't match in your server. 

I would extract this out to it's own describe block, separate from this endpoint's tests. 

Apart from that, we don't have other errors to consider 👍

File: app.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const express = require("express");
+const { getTopics } = require("./controller");
+const app = express();
+
+app.get('/api/topics', getTopics)
+app.all(`/*`, (req, res) => {res.status(404).send({ msg: "Data Not Found"})})
```
Comment (Dutton1990) [inline app.js:6]: nice implementation!

File: __tests__/app.test.js (hunk 3 above)
Comment (denton909) [reply __tests__/app.test.js:32]: I have updated the tests suite to do this. Am not sure how i show it to you with out doing a new pull request. Are you able to see the changes as i have pushed to nc-news-2?

**Pull request merged**